### PR TITLE
[#11858] Instructor viewing results: 'Other teams in the same section' path is causing timeout

### LINKED ISSUE #11858: Instructor viewing results: 'Other teams in the same section' path is causing timeout
Reported by a user
V8.16.1

A user reported (and I verified) that the results for questions having a certain feedback path is not loading, causing a timeout in the end.

image

Q1 (rubric) and Q2 (essay) have this path and are not loading. 
image

Q2 (rubric) has a different path (given below) and is loading.

image

Perhaps `Other teams in the same section` (a rarely used path) is buggy?

PS: the error message in the log
>Error Message: DeadlineExceededException caught by WebApiServlet
caused by teammates.common.exception.DeadlineExceededException: null
at teammates.common.util.RequestTracer.checkRemainingTime(RequestTracer.java:59)
at teammates.logic.core.FeedbackResponsesLogic.buildResultsBundle(FeedbackResponsesLogic.java:437)
at teammates.logic.core.FeedbackResponsesLogic.getSessionResultsForCourse(FeedbackResponsesLogic.java:479)
at teammates.logic.api.Logic.getSessionResultsForCourse(Logic.java:1262)
at teammates.ui.webapi.GetSessionResultsAction.execute(GetSessionResultsAction.java:76)
at teammates.ui.webapi.GetSessionResultsAction.execute(GetSessionResultsAction.java:15)
at teammates.ui.servlets.WebApiServlet.invokeServlet(WebApiServlet.java:65)
at teammates.ui.servlets.WebApiServlet.doGet(WebApiServlet.java:38)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)

### PR DESCRIPTION
**Pull request by xyliew25**
Fixes #11858

**Outline of Solution**
Timeout is believed to be caused by the building of missing responses for `Other teams in the same section`, where a query is made instead of utilizing the course roster provided. Hence, the fix is just to utilize the course roster.

### CODE REVIEW

**Comment by damithc** [comment]
Thanks for the PR @xyliew25 
Any performance comparison data for the given student list?

File: src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java [hunk 1]
```diff
@@ -364,12 +364,19 @@ public Map<String, String> getRecipientsOfQuestion(
...
+                }
                 teamToTeamMembersTable = CourseRoster.buildTeamToMembersTable(teamStudents);
             } else {
-                if (courseRoster == null) {
-                    teamStudents = studentsLogic.getStudentsForCourse(question.getCourseId());
+                if (generateOptionsFor == FeedbackParticipantType.TEAMS_IN_SAME_SECTION) {
+                    final String finalGiverSection = giverSection;
+                    teamStudents = courseRoster.getStudents().stream()
+                            .filter(studentAttributes -> studentAttributes.getSection()
+                                    .equals(finalGiverSection)).collect(Collectors.toList());
```
Comment (wkurniawan07) [inline src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java:379]: Minor readability fix:
```java
teamStudents = courseRoster.getStudents().stream()
        .filter(student -> student.getSection().equals(finalGiverSection))
        .collect(Collectors.toList());
```

**Review by wkurniawan07 — COMMENTED** [review]
The fix definitely works; can tell by just looking at the code change. Minor suggestion left. Let's get this ready asap.

**Comment by xyliew25** [comment]
@damithc performance comparison data for the given student list:
image
image

**Comment by damithc** [comment]
> @damithc performance comparison data for the given student list: image image

Nice. Good work @xyliew25

**Review by wkurniawan07 — APPROVED** [review]

**Pull request merged**